app/services/location_comparer.rb

```diff
@@ -0,0 +1,31 @@
+class LocationComparer
```

Comment:
Yeah, I thought you might object to this class.

`different?`, `different_zip?`, 'different_city?`, and `different_state?` were all methods on `LocationUpdater` before I extracted this class. They were the best way I had come up with to make the old tests pass. It's possible that there's a simpler way to do it and that it would obviate the need for a separate class, but I think we need to nail down the rules to figure that out. There is one big advantage to doing it this way (besides blind observance of single-responsibility principle), which is that it makes testing much easier. This way we have all the rules about location comparison in one simple class with no dependencies and can test a table of values to ensure that we get the correct results. Trying to test the location comparison rules in LocationUpdater requires a lot of setup for each test. If the rules get more complicated, this class will start to offer more value.

Re: using LocationComparable instead of LocationComparer, I'm open to the idea but don't yet understand the advantage. Seems like it would add even more complexity -- what problem would it solve? Also, as I have written the rules now, new location and old location are treated slightly differently, which I think would be harder to do with LocationComparable. Then again, maybe it wasn't necessary to treat the new and old values differently...